src/components/course/EnrollButton.jsx

```diff
@@ -131,176 +71,124 @@ export default function EnrollButton() {
     () => hasCourseStarted(start),
     [start],
   );
-
-  const isUserEnrolled = useMemo(
-    () => isUserEnrolledInCourse({ userEnrollments, key }),
+  const userEnrollment = useMemo(
+    () => findUserEnrollmentForCourse({ userEnrollments, key }),
     [userEnrollments, key],
   );
+  const enrollmentUrl = useCourseEnrollmentUrl({
+    catalogList,
+    enterpriseConfig,
+    key,
+    location,
+    offers,
+    sku,
+    subscriptionLicense,
+  });
+
+  const shouldUpgradeUserEnrollment = useMemo(
```

Comment:
I feel like we may be over-optimizing by using `useMemo` everywhere (I definitely do it too). This isn't a high cost operation that's being performed.

Reply:
Food for thought here: https://blog.logrocket.com/rethinking-hooks-memoization/

Reply:
Good call out. I agree we're probably overusing `useMemo` throughout the app at this point. Thanks for sharing that article; good read!